File: src/main/java/com/nexomc/example_mechanic/ExampleMechanic.java
package com.nexomc.example_mechanic;

import com.nexomc.nexo.mechanics.Mechanic;
import com.nexomc.nexo.mechanics.MechanicFactory;
import org.bukkit.configuration.ConfigurationSection;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ExampleMechanic extends Mechanic {

    @NotNull public final String something;
    public final boolean bool;

    public ExampleMechanic(@Nullable MechanicFactory factory, @NotNull ConfigurationSection section) {
        super(factory, section);
        something = section.getString("something", "nothing");
        bool = section.getBoolean("bool", false);
    }
}

File: src/main/kotlin/com/nexomc/mechanic_example/ExampleMechanic.kt
package com.nexomc.mechanic_example

import com.nexomc.nexo.mechanics.Mechanic
import org.bukkit.configuration.ConfigurationSection

class ExampleMechanic(factory: ExampleMechanicFactory, section: ConfigurationSection) : Mechanic(factory, section) {
    val something: String = section.getString("something") ?: "nothing"
    val bool: Boolean = section.getBoolean("bool")
}